aws/src/main/java/org/apache/iceberg/aws/s3/S3FileIO.java

```diff
@@ -155,6 +190,31 @@ public void deleteFiles(Iterable<String> paths) throws BulkDeletionFailureExcept
     }
   }
 
+  private void tagFileToDelete(String path, Set<Tag> deleteTags) throws S3Exception {
+    S3URI location = new S3URI(path);
```

Comment:
can we wire in s3BucketToAccessPointMapping, to use access point in tag manipulation. 

Adding Tag's via access point is an allowed operation : https://docs.aws.amazon.com/AmazonS3/latest/userguide/access-points-usage-examples.html#add-tag-set-ap i.e both GetObjectTagging as well as PutObjectTagging operations are allowed

Reply:
Nice catch, missed on added this after merging master.